File: get_bj.py
import urllib.request

stocks = [('普适导航', '831330'), ('圣博润', '430046')]
for name, code in stocks:
    try:
        url = f'http://hq.sinajs.cn/list=ob{code}'
        req = urllib.request.Request(url, headers={'Referer': 'http://finance.sina.com.cn'})
        with urllib.request.urlopen(req, timeout=10) as r:
            data = r.read().decode('gbk')
            parts = data.split('"')[1].split(',')
            price = float(parts[3])
            prev = float(parts[2])
            pct = (price - prev) / prev * 100
            print(f'{name}|{code}|{price}|{pct:.2f}')
    except Exception as e:
        print(f'{name}|{code}|ERROR|{e}')

File: get_bj.js
const http = require('http');

function fetchStock(code, prefix) {
  return new Promise((resolve) => {
    const options = {
      hostname: 'hq.sinajs.cn',
      path: '/list=' + prefix + code,
      headers: {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36',
        'Referer': 'https://finance.sina.com.cn/'
      }
    };
    http.get(options, (res) => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => {
        try {
          const m = d.match(/"([^"]+)"/);
          if (m) {
            const parts = m[1].split(',');
            const price = parseFloat(parts[3]);
            const prevClose = parseFloat(parts[2]);
            const chgPct = prevClose > 0 ? ((price - prevClose) / prevClose * 100).toFixed(2) : '0.00';
            resolve({ code, name: parts[0], price, prevClose, changePct: chgPct });
          } else {
            resolve({ code, error: 'no match', raw: d.substring(0, 200) });
          }
        } catch(e) {
          resolve({ code, error: e.message, raw: d.substring(0, 200) });
        }
      });
    }).on('error', e => resolve({ code, error: e.message }));
  });
}

async function main() {
  const results = await Promise.all([
    fetchStock('831330', 'sz'),  // 普适导航
    fetchStock('430046', 'bj'),  // 圣博润
  ]);
  console.log(JSON.stringify(results));
}
main().catch(console.error);
